internal/daemonset/daemonset.go

```diff
@@ -472,3 +493,20 @@ func MakeUnloadCommand(spec kmmv1beta1.ModprobeSpec, modName string) []string {
 
 	return append(unloadCommandShell, unloadCommand.String())
 }
+
+func getModulesOrderAnnotationValue(mld *api.ModuleLoaderData) string {
+	modulesNames := mld.Modprobe.ModulesLoadingOrder
+	softDepData := ""
+	for i := 0; i < len(modulesNames); i++ {
+		if i == len(modulesNames)-1 {
+			break
+		}
+		line := prepareSoftDepLine(modulesNames[i], modulesNames[i+1])
+		softDepData = softDepData + line
+	}
+	return softDepData
+}
+
+func prepareSoftDepLine(dependendModuleName, moduleName string) string {
+	return fmt.Sprintf("softdep %s pre: %s\n", dependendModuleName, moduleName)
+}
```

Comment:
Could we use a Go template here instead?

Reply:
We can, but i am not sure if that will male the code more readable , or reduce the number of code line. We will need to use range inside template definition, and we will still need to use the loop over the modulesName to prepare the input data for templates